pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -201,8 +201,18 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	}
 	allErrors = append(allErrors, metrics.ValidateShowHiddenMetricsVersion(kc.ShowHiddenMetricsForVersion)...)
 
+	return utilerrors.NewAggregate(allErrors)
+}
+
+// ValidateKubeletLoggingConfiguration validate kubelet logging part config
+func ValidateKubeletLoggingConfiguration(kc *kubeletconfig.KubeletConfiguration) error {
+	allErrors := []error{}
+	logOption := &logs.Options{Config: kc.Logging}
+
 	if errs := logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")); len(errs) > 0 {
 		allErrors = append(allErrors, errs.ToAggregate().Errors()...)
 	}
+
+	logOption.Apply()
```

Comment:
Apply should not be put in ValidateKubeletLoggingConfiguration function as current there is a convention that flags are handled with 3 separate steps (Bind, Validate, Apply). Having a Apply in Validate breaks this convention. Instead of having separate function could you simply code and inline it in `cmd/kubelet/app/server.go`